helix-term/src/commands.rs

```diff
@@ -1193,6 +1211,101 @@ fn search_selection(cx: &mut Context) {
     search_next(cx);
 }
 
+fn global_search(cx: &mut Context) {
+    let (all_matches_sx, all_matches_rx) =
+        tokio::sync::mpsc::unbounded_channel::<(usize, PathBuf)>();
+    let prompt = ui::regex_prompt(
+        cx,
+        "global search:".to_string(),
+        move |_view, _doc, registers, regex| {
+            registers.write('/', vec![regex.as_str().to_string()]);
+        },
+        move |_view, regex| {
+            if let Ok(matcher) = RegexMatcher::new_line_matcher(regex.as_str()) {
+                let searcher = SearcherBuilder::new()
+                    .binary_detection(BinaryDetection::quit(b'\x00'))
+                    .build();
+
+                let search_root =
+                    std::env::current_dir().expect("Global Search: Failed to get current dir");
+                WalkBuilder::new(search_root).build_parallel().run(|| {
+                    let mut searcher_cl = searcher.clone();
+                    let matcher_cl = matcher.clone();
+                    let all_matches_sx_cl = all_matches_sx.clone();
```

Comment:
Why clone here? Can't you make it a move closure? `.run(move || {`

Reply:
[WalkParallel::run](https://docs.rs/ignore/0.4.18/ignore/struct.WalkParallel.html#method.run) takes a callback that returns a `Box<dyn FnMut(Result<DirEntry, Error>) -> WalkState + Send + 's>`, which would capture `searcher`, `matcher`, `all_matches_sx`. So I think the clone is necessary here.